src/qmapshack/gis/rte/router/CRouterRoutino.cpp

```diff
@@ -300,13 +327,28 @@ void CRouterRoutino::calcRoute(const IGisItem::key_t& key)
         {
             throw QString();
         }
-
-        Routino_Database * data = (Routino_Database*)comboDatabase->currentData(Qt::UserRole).toULongLong();
+        int res = 0;
+        QVariantMap map = comboDatabase->currentData(Qt::UserRole).toMap();
+        Routino_Database * data = (Routino_Database*)(map["db"].toULongLong());
         if(nullptr == data)
         {
             throw QString();
         }
 
+        // load xml profiles if it is new before setting the profile!
+        QString prefixpath = map["prefixpath"].toString();
+        if( currentPrefix != prefixpath)
+        {
+            currentPrefix = prefixpath;
+            printf("cR CGIR PrefixPath %s",prefixpath.toStdString().data());
+            res = Routino_ParseXMLProfiles(prefixpath.toStdString().data());
+            if(res)
```

Comment:
Did you make sure this will remove the old profiles and install new ones? Or does it just sum up profiles?

Reply:
I have checked - the parseProfileXML routine deletes the existing profiles data before handling the new one.  I have checked operation in working code as well.

Other question regarding the PR template - where is this explained in the wiki - cant find it.  Some of it I didnt understand.    I have cleaned up the code, including variable naming (sorry I missed that).  I think I now comply with the coding guidelines.
Also your comment on changelog the ticket number.  Is it the Issues ticket number on my initial request? And where do I explain in detail the changelog entry - in git?

I have appreciated your comments.

Reply:
No it's not written explicitly in the wiki that you shouldn't delete parts of the issue and PR templates. But common sense should tell you to fill it out without removing sections. 

Yes, your initial ticket is the ticket that should have been linked to that PR. There is a line in the template where simply have to add the number. But you erased it.

The changelog is the changelog.txt in the source tree. It's simply copying the title of the ticket into the upmost section. And it wouldn't hurt to think about the title. Does it still reflect the content of the change? If not change the title.  The entries in the changelog are kind of ordered. Make sure to place it into the order. By that it's easier to find for everyone.

Another point is providing a simple smoke test. These PR are public and others not deeply involved on the topic might want to test. They will if they know how to do. And I know what you did test and might think of tests you didn't do.

You can find the template in `.github/PULL_REQUEST_TEMPLATE.md` in the source tree.